imap_processing/idex/idex_packet_parser.py

```diff
@@ -133,35 +161,335 @@ def __init__(self, header_packet):
+        ].raw_value
+        trigger_notes_dict[
+            "low_sample_trigger_mode"
+        ] = "Low sample trigger mode (0=disabled, 1=enabled)"
+        trigger_dict["tof_low_trigger_mode"] = packet.data[
+            "IDX__TXHDRLSTRIGMODE"
+        ].raw_value
+        trigger_notes_dict[
+            "tof_low_trigger_mode"
+        ] = "TOF Low trigger mode (0=disabled, 1=enabled)"
+        trigger_dict["tof_mid_trigger_mode"] = packet.data[
+            "IDX__TXHDRMGTRIGMODE"
+        ].raw_value
+        trigger_notes_dict[
+            "tof_mid_trigger_mode"
+        ] = "TOF Mid trigger mode (0=disabled, 1=enabled)"
+        trigger_dict["tof_high_trigger_mode"] = packet.data[
+            "IDX__TXHDRHGTRIGMODE"
+        ].raw_value
+        trigger_notes_dict[
+            "tof_high_trigger_mode"
+        ] = """TOF Mid trigger mode (0=disabled, 1=threshold mode, 2=single pulse
+               mode, 3=double pulse mode)"""
+
+        detector_sensor_voltage = packet.data["IDX__TXHDRHVPSHKCH01"].raw_value
+        trigger_dict["detector_voltage"] = (
+            detector_sensor_voltage >> 10 & self.TWELVE_BIT_MASK
+        )
+        trigger_notes_dict[
+            "detector_voltage"
+        ] = "Last measurement in raw dN for processor board signal: Detector Voltage"
+        trigger_dict["sensor_voltage"] = (
+            detector_sensor_voltage >> 10 & self.TWELVE_BIT_MASK
+        )
+        trigger_notes_dict[
+            "sensor_voltage"
+        ] = "Last measurement in raw dN for processor board signal: Sensor Voltage"
+        target_reflectron_voltage = packet.data["IDX__TXHDRHVPSHKCH23"].raw_value
+        trigger_dict["target_voltage"] = (
+            target_reflectron_voltage >> 10 & self.TWELVE_BIT_MASK
+        )
+        trigger_notes_dict[
+            "target_voltage"
+        ] = "Last measurement in raw dN for processor board signal: Target Voltage"
+        trigger_dict["reflectron_voltage"] = (
+            target_reflectron_voltage >> 10 & self.TWELVE_BIT_MASK
+        )
+        trigger_notes_dict[
+            "reflectron_voltage"
+        ] = "Last measurement in raw dN for processor board signal: Reflectron Voltage"
+        rejection_voltage_detector_current = packet.data[
+            "IDX__TXHDRHVPSHKCH45"
+        ].raw_value
+        trigger_dict["rejection_voltage"] = (
+            rejection_voltage_detector_current >> 10 & self.TWELVE_BIT_MASK
+        )
+        trigger_notes_dict[
+            "rejection_voltage"
+        ] = "Last measurement in raw dN for processor board signal: Rejection Voltage"
+        trigger_dict["detector_current"] = (
+            rejection_voltage_detector_current >> 10 & self.TWELVE_BIT_MASK
+        )
+        trigger_notes_dict[
+            "detector_current"
+        ] = "Last measurement in raw dN for processor board signal: Detector Current"
+
+        return trigger_dict, trigger_notes_dict
 
     def _calc_impact_time(self, packet):
```

Comment:
could I use this function in my PR to convert epoch time to unix time? or can we create a utils folder under `imap_processing` and move this function there since this seems like a common function that we all could benefit from?

Every packet that we get has time in epoch. It's called SHCOARSE or MET.

Reply:
Yes that sounds good! I guess it all must be in seconds before 2012?

Reply:
I will do that in future PR because I don't want extend both PRs. It should be small change.